src/comments.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import json
+import argparse
+import os
+import requests
+import sys
+from xmlschema import XMLSchema
+from pprint import pprint
+import os
+from unidiff import PatchSet
+
+try:
+    from .providers import find_ci_provider, gitroot, gitdiff
+except ImportError:
+    from providers import find_ci_provider, gitroot, gitdiff
+
+
+try:
+    from .utils import EnvDefault, valid_repo_name, valid_token, valid_pr, valid_file, valid_sha, pick
+except ImportError:
+    from utils import EnvDefault, valid_repo_name, valid_token, valid_pr, valid_file, valid_sha, pick
+
+
+GITHUB_BASE = os.environ.get("GITHUB_BASE", "https://api.github.com")
+GITHUB_PR_COMMENTS_URL = GITHUB_BASE + "/repos/{repo_name}/pulls/{pull_request}/comments?access_token={token}"
+
+ACTIVE_CI = find_ci_provider()
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+def format_filename(name):
+    root = gitroot()
+    for f in [os.path.join(root, name), name]:
+        full_name = os.path.abspath(f)
+        if os.path.exists(full_name):
+            return full_name[len(root)+1:]
+
+    return name
+
+def change_in_diff(filename, row, diff):
+    def file_in_diff(fn):
+        if filename == fn.path:
+            return True
+        elif os.path.join("a", filename) == fn.source_file:
+            return True
+        elif os.path.join("b", filename) == fn.target_file:
+            return True
+        return False
+
+    def line_in_hunk():
+        for ps in diff:
+            if file_in_diff(ps):
+                print(filename)
+                for hunk in ps:
+                    r = range(hunk.target_start, hunk.target_start + hunk.target_length + 1)
+                    if row in r:
+                        return True
+        return False
+
+    b = line_in_hunk()
+    if not b:
+        print(f"LINE: {row} NAME: {filename}")
```

Comment:
src/comments.py:62:5: F841 local variable 'link_matcher' is assigned to but never used